content/reference/apis/config_var/kv.md

```diff
@@ -0,0 +1,261 @@
+API will omit any expired keys from the response.
+
+You can choose one of two ways to specify when a key should expire:
+
+- Set its "expiration", using an absolute time specified in a number of
+  [seconds since the UNIX epoch](https://en.wikipedia.org/wiki/Unix_time). For
+  example, if you wanted a key to expire at 12:00AM UTC on April 1, 2019, you
+  would set the key's expiration to 1554076800.
+- Set its "expiration TTL" (time to live), using a relative number of seconds
+  from the current time. For example, if you wanted a key to expire 10 minutes
+  after creating it, you would set its expiration TTL to 600.
+
+Both of these options are usable when writing a key inside a Worker or when
+writing keys using the API.
+
+Note that expiration times of less than 60 seconds in the future or
+expiration TTLs of less than 60 seconds are not supported at this time.
+
+### Creating expiring keys
+
+We talked about the basic form of the `put` method above, but this call also
+has an optional third parameter. It accepts an object with optional fields
+that allow you to customize the behavior of the `put`. In particular, you can
+set either `expiration` or `expirationTtl`, depending on how you would like
+to specify the key's expiration time. In other words, you'd run one of the
+two commands below to set an expiration when writing a key from within a
+Worker:
+
+`NAMESPACE.put(key, value, {expiration: secondsSinceEpoch})`
+
+`NAMESPACE.put(key, value, {expirationTtl: secondsFromNow})`
+
+These assume that `secondsSinceEpoch` and `secondsFromNow` are variables
+defined elsewhere in your Worker code.
+
+You can also [write with an expiration on the command line via
+Wrangler](/tooling/wrangler/kv_commands/#kv-key) or [via the
+API](https://api.cloudflare.com/#workers-kv-namespace-write-key-value-pair).
+
+# Reading key-value pairs
+
+To get the value for a given key, you can call the `get` method on any
+namespace you've bound to your script:
+
+`NAMESPACE.get(key)`
+
+The method returns a promise you can `await` to get the value. If the key
+is not found, the promise will resolve with the literal value `null`.
+
+Here's an example of reading a key from within a Worker:
+
+```js
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+
+async function handleRequest(request) {
+  const value = await FIRST_KV_NAMESPACE.get("first-key")
+  if (value === null) {
+    return new Response("Value not found", {status: 404})
+  }
+
+  return new Response(value)
+}
+```
+
+You can also [read key-value pairs from the command line with
+wrangler](/tooling/wrangler/kv_commands/#kv-key).
+
+Finally, you can also [read from the
+API](https://api.cloudflare.com/#workers-kv-namespace-read-key-value-pair).
+
+## Types
+
+You can pass an optional `type` parameter to the `get` method as well:
+
+`NAMESPACE.get(key, type)`
+
+The `type` parameter can be any of:
+
+- `"text"`: (default) a string
```

Comment:
in the `NAMESPACE.put` section above, we don't link these parameter types out to mdn. can we link them here too?